src/availability.rs

```diff
@@ -280,9 +333,9 @@ where
 mod test {
```

Comment:
This test block requires the `"no-storage"` feature in order to build correctly.

```suggestion
#[cfg(all(test, feature = "no-storage"))]
mod test {
```

Reply:
Careful here, it wouldn't let me select the `cfg` line for the `suggestion` for some reason.

Reply:
Oh yeah, we generally just require `--all-features` for building test code. Cuz even if we made it work, building without `--all-features` would probably not be what you want as it would silently skip a bunch of tests

Reply:
That makes sense.  It was just something that the `rust-analyzer` was complaining about.